### PR TITLE
Add AtLeastOneOf validation, code refactoring

### PR DESCRIPTION
**Pull request by akonshin**
ticket for reference: https://trello.com/c/UEHESspu/29-atleastoneof-validation

### CODE REVIEW

File: copy_cat/validators/validation_conditions/at_least_one_of_validator.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from copy_cat.models.error import Error
+from copy_cat.utils import get_test_data_object
+
+
+class AtLeastOneOfValidator:
+    def __init__(self):
+        self.errors = []
+
+    def validate(self, validations: list, test_data: list, location: list):
```
[the remark this answers is not in the document]
Comment (akonshin) [reply copy_cat/validators/validation_conditions/at_least_one_of_validator.py:9]: agree

File: copy_cat/validators/choices_validator.py [hunk 2]
```diff
@@ -7,18 +8,17 @@ def __init__(self):
 
     def validate(self, design_object, test_data_object):
         self.validate_choices(design_object, test_data_object)
+        return self.errors
```
[the remark this answers is not in the document]
Comment (akonshin) [reply copy_cat/validators/choices_validator.py:11]: It is okay to me, it validates something and returns the result of validation ...

File: copy_cat/validators/choices_validator.py [hunk 3]
```diff
@@ -7,18 +8,17 @@ def __init__(self):
...
-                self.errors.append({
-                    "fieldName": test_data_object['name'],
-                    "fieldPath": design_object['location'],
-                    "xpath": test_data_object["location"],
-                    "error": error_message,
-                })
+                self.errors.append(Error(fieldName=test_data_object.name,
+                                         fieldPath=design_object['location'],
+                                         xpath=test_data_object.location,
+                                         error=error_message))
```
[the remark this answers is not in the document]
Comment (akonshin) [reply copy_cat/validators/choices_validator.py:24]: I'll update naming

File: copy_cat/validators/validator.py [hunk 4]
```diff
@@ -1,39 +1,39 @@
...
         self.data_type_validator = DataTypeValidator()
         self.choices_validator = ChoicesValidator()
         self.length_validator = LengthValidator()
+        self.validation_conditions_validator = ValidationConditionsValidator()
 
-    def validate(self, test_data, schema):
+    def validate(self, schema, test_data):
         for ind, t in enumerate(test_data):
-            location = t.get('location')[9:]
+            location = t.location[9:]
```
[the remark this answers is not in the document]
Comment (akonshin) [reply copy_cat/validators/validator.py:24]: I'll take a look

**Review by pashura — APPROVED** [review]

**Pull request merged**